video/video_library_interface_review.rb

```diff
@@ -0,0 +1,184 @@
+# DO NOT MERGE - this file for comments only
+
+def analyze_labels_gcs path:
+  # [START analyze_labels_gcs]
+  # path = "Path to a video file on Google Cloud Storage: gs://bucket/video.mp4"
+
+  require "google/cloud/video_intelligence/v1beta1"
+
+  video_client = Google::Cloud::VideoIntelligence::V1beta1::VideoIntelligenceServiceClient.new
+  features     = [:LABEL_DETECTION]
+
+  # Register a callback during the method call
+  operation = video_client.annotate_video path, features do |operation|
+    raise operation.results.message? if operation.error?
+    puts "Finished Processing."
+
+    # first result is retrieved because a single video was processed
+    annotation_result = operation.results.annotation_results.first
+
+    annotation_result.label_annotations.each do |label_annotation|
+      puts "Label description: #{label_annotation.description}"
+      puts "Locations:"
+
+      label_annotation.locations.each do |location|
+        if location.level == :VIDEO_LEVEL
+          puts "Entire video"
+        else
+          segment          = location.segment
+          start_in_seconds = segment.start_time_offset / 1000000.0
+          end_in_seconds   = segment.end_time_offset / 1000000.0
+
+          puts "#{start_in_seconds} through #{end_in_seconds}"
+        end
+      end
+    end
+  end
+
+  puts "Processing video for label annotations:"
+  operation.wait_until_done!
+  # [END analyze_labels_gcs]
+end
+
+def analyze_labels_local path:
+  # [START analyze_labels_local]
+  # path = "Path to a local video file: path/to/file.mp4"
+
+  require "base64"
+  require "google/cloud/video_intelligence/v1beta1"
+
+  video_client  = Google::Cloud::VideoIntelligence::V1beta1::VideoIntelligenceServiceClient.new
+  features      = [:LABEL_DETECTION]
+  video_file    = File.read path
+  encoded_video = Base64.encode64 video_file
+
+  # Register a callback during the method call
+  operation = video_client.annotate_video nil, features, input_content: encoded_video do |operation|
```

Comment:
## BLOCKER!

That `nil` argument needs to go away 😛

Reply:
If you want a named param for this, then...

With GCS path

```ruby
operation = video_client.annotate_video features, gcs_path: "gcs://foo"
```

```ruby
operation = video_client.annotate_video features, video_content: encoded_video
```

NOTE: The AnnotateVideoRequest Proto should have a ONE OF for input_uri vs input_content

OK... looking at the proto, I think `input_uri` and `input_content` are satisfactory and clear 😄

Reply:
TODO: short-term -- we can mark `input_uri` optional in GAPIC config. longer-term -- the API should maybe consider using a oneof for `{input_uri, input_content}`?